uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java

```diff
@@ -0,0 +1,124 @@
+                                            "attr_name_0",
+                                            "attr_name_1",
+                                            "attr_name_2",
+                                            "attr_name_3"
+                                        })));
+
+        CURRENT_ATTRIBUTE_VALUES =
+                Collections.unmodifiableList(
+                        new ArrayList<>(Arrays.asList(new String[] {"attr_value_0"})));
+    }
+
+    @Before
+    public void setup() throws Exception {
+        userAccountHelper = new UserAccountHelper();
+        userAccountHelper.setAccountEditAttributes(generateAccountEditPreferences());
+        accountDao = mock(ILocalAccountDao.class);
+        accountPerson = mock(ILocalAccountPerson.class);
+        when(accountPerson.getName()).thenReturn(TEST_USER_NAME);
+        when(accountPerson.getId()).thenReturn(10000l);
+        when(accountPerson.getAttributeValue("loginToken")).thenReturn(TEST_USER_LOGIN_TOKEN);
+        when(accountDao.getCurrentAttributeNames()).thenReturn(CURRENT_ATTRIBUTE_NAMES);
+        when(accountPerson.getAttributeValues(TEST_ATTRIBUTE_NAME))
+                .thenReturn(CURRENT_ATTRIBUTE_VALUES);
+        when(accountDao.getPerson(TEST_USER_NAME)).thenReturn(accountPerson);
+        userAccountHelper.setLocalAccountDao(accountDao);
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testGetNewAccountForm() {
+        PersonForm personForm = userAccountHelper.getNewAccountForm();
+        Map<String, StringListAttribute> personAttributeList = personForm.getAttributes();
+        assertNotNull(personForm);
+        assertEquals(4, personAttributeList.size());
+    }
+
+    @Test
+    public void testGetForm() {
+        PersonForm personForm = userAccountHelper.getForm(TEST_USER_NAME);
+        assertEquals(personForm.getId(), 10000l);
+        assertEquals(personForm.getUsername(), TEST_USER_NAME);
+        // validate person attributes
+        Map<String, StringListAttribute> personAttributes = personForm.getAttributes();
+        assertNotNull(personAttributes);
+        StringListAttribute attributeValues = personAttributes.get(TEST_ATTRIBUTE_NAME);
+        List<String> values = attributeValues.getValue();
+        assertEquals(values.get(0), CURRENT_ATTRIBUTE_VALUES.get(0));
+    }
+
+    @Test
+    public void testValidateLoginToken() {
+        boolean isTokenValid =
+                userAccountHelper.validateLoginToken(TEST_USER_NAME, TEST_USER_LOGIN_TOKEN);
+        assertTrue(isTokenValid);
+    }
+
+    /** @return */
+    private List<Preference> generateAccountEditPreferences() {
+        List<Preference> accountEditPreferences = new ArrayList<>();
+        Preference preference = new Preference();
+        preference.setName("pref_name_0");
+        preference.setDescription("pref_description_0");
+        accountEditPreferences.add(preference);
+        return accountEditPreferences;
+    }
+}
```

Comment:
There looks to be several methods that could be added to this class such as canDeleteUser and canEditUser (at least up to the first conditional), createPassword, getNewAccountForm, etc.